daemon/api.go

```diff
@@ -1782,6 +1782,44 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
 	}, nil)
 }
 
+func convertBuyError(err error) Response {
+	switch err {
+	default:
+		return InternalError("%v", err)
+	case store.ErrInvalidCredentials:
+		return Unauthorized(err.Error())
+	case store.ErrUnauthenticated:
+		return SyncResponse(&resp{
```

Comment:
(and yes, I know there already are a bunch of these, so if you want to leave it I'll do it in a branch all of its own)